examples/async-data/src/App.js

```diff
@@ -58,37 +54,22 @@ class App extends Component {
     super();
 
     // A basic starting plotly.js figure object. Instead of assigning
-    const figure = {
+    const graphDiv = {
       data: [{type: 'scatter'}],
       layout: {title: 'Room readings'},
     };
 
```

Comment:
I'm not sure about this, when the plot component updates, it sets graphDiv to the actual dom element:
https://github.com/plotly/react-plotly.js/blob/master/src/factory.js#L167

it's that actual dom element that we're passing around and deriving all our values from, so making it seem like it is not the case is confusing to me

@bpostlethwaite is this not the case, am I missing something here?

Reply:
Also, I don't think I understand how that state.figure used to work and update, what was making it update?

Reply:
Plotly.js would basically copy the `figure.data` and `figure.layout` references into `graphDiv` and pass it to the Editor via the Hub... Now it's just more explicit that the App has a handle on `graphDiv`: it partially-constructs it (i.e. gives it `data` and `layout`) and then Plotly.js adds all the other stuff it needs to  (basically by overwriting it every time it updates)

Reply:
but to me it seems like the graphDiv that plotly.js passes is the actual dom div element.. when looking at the code

Reply:
https://github.com/plotly/react-plotly.js/blob/master/src/factory.js#L189

Reply:
It is, but that dom div element has `data` and `layout` and `_fullLayout` etc attached to it as well...

Reply:
yes I know :) the code here hides it a little, makes it seem like graphDiv is just an object, when that object is later changed to an actual dom node from which data, layout and everything else's derived. 

I'm ok with how it is used in instantiation here, but let's say I'd prefer we don't rename it like you're suggesting here: https://github.com/plotly/react-plotly.js-editor/issues/178

I think that editor users will most likely also be familiar with plotly.js and won't be surprised about our use of graphDiv and the fact that its a dom node that has all these properties

Reply:
maybe we could add a note about it in here

Reply:
Let's continue the conversation in #178 :)

Reply:
Re the code here, we could also just have an `initialFigure = {data, layout}` type thing, with the initial value of `graphDiv` as `null`... I had it this way earlier and it worked as well, if we want to make it really explicit what's happening.

Reply:
what note would you add?

Reply:
that graphDiv is later on replaced by a full dom node with all the other properties of plotly.js

Reply:
> Re the code here, we could also just have an initialFigure = {data, layout} type thing, with the initial value of graphDiv as null
I think I prefer that

Then no note is really needed :)

Reply:
I added a comment per your suggestion. The reason I removed the `initialFigure` thing is that in `render()` it looked really weird to have `<Plot data={this.initialFigure.data} layout={this.initialFigure.layout} ... />` because it seemed like it would never get new data.